Add integration test for volume controller startup.

jsafrane opened the pull request:
Tests #28002 with real etcd (unit tests have a fake one with different behavior).

@kubernetes/sig-storage

jingxu97 commented on test/integration/persistent_volumes_test.go:
```diff
@@ -596,14 +596,121 @@ func TestPersistentVolumeMultiPVsPVCs(t *testing.T) {
+
+	// wait for at least two sync periods for changes. No volume should be
+	// Released and no claim should be Lost during this time.
+	timer := time.NewTimer(2 * syncPeriod)
```
Just wondering whether two sync periods definitely enough? (I am not very familiar with this part of code)

jsafrane replied:
Each PV and PVC evaluated every sync period. Assuming that the PVs/PVCs are bound together, this evaluation checks that a PV still has corresponding PVC and a vice versa. When one of this pair is missing, PV gets Released or PVC gets Lost.

So, let's wait for two sync periods to make sure at least one check happened (and finished) for sure.

jingxu97 commented on test/integration/persistent_volumes_test.go:
```diff
@@ -596,14 +596,121 @@ func TestPersistentVolumeMultiPVsPVCs(t *testing.T) {
+			if !ok {
+				continue
+			}
+			if volume.Status.Phase == api.VolumeReleased {
```
If the cache is not initialized properly, the Phase will change to released? Any other Phase possible?

jsafrane replied:
The bug in #28002 was that some PVs could get `Released` and some PVCs `Lost`.  I reorganized the test a bit to catch also other phases as error now.

jsafrane commented:
Rebased and reorganized to report *all* phase changes as error. PVs and PVCs must stay `Bound` for duration of the test.

jsafrane commented:
Rebased

jsafrane committed:
Add integration test for volume controller startup.

Tests #28002 with real etcd (unit tests have a fake one with different
behavior).

jsafrane commented:
rebased

jingxu97 commented:
@ixdy, can we remove do-not-merge label? Thanks!
